src/profile-logic/marker-schema.js

```diff
@@ -0,0 +1,217 @@
+      {
+        key: 'filename',
+        label: 'Filename',
+        format: 'file-path',
+        searchable: true,
+      },
+    ],
+  },
+  {
+    name: 'MediaSample',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      {
+        key: 'sampleStartTimeUs',
+        label: 'Sample start time',
+        format: 'microseconds',
+      },
+      {
+        key: 'sampleEndTimeUs',
+        label: 'Sample end time',
+        format: 'microseconds',
+      },
+    ],
+  },
+  {
+    name: 'Styles',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      {
+        key: 'elementsTraversed',
+        label: 'Elements traversed',
+        format: 'integer',
+      },
+      { key: 'elementsStyled', label: 'Elements styled', format: 'integer' },
+      { key: 'elementsMatched', label: 'Elements matched', format: 'integer' },
+      { key: 'stylesShared', label: 'Styles shared', format: 'integer' },
+      { key: 'stylesReused', label: 'Styles reused', format: 'integer' },
+    ],
+  },
+  {
+    name: 'PreferenceRead',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      { key: 'prefName', label: 'Name', format: 'string' },
+      { key: 'prefKind', label: 'Kind', format: 'string' },
+      { key: 'prefType', label: 'Type', format: 'string' },
+      { key: 'prefValue', label: 'Value', format: 'string' },
+    ],
+  },
+  {
+    name: 'Invalidation',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      { key: 'url', label: 'URL', format: 'url' },
+      { key: 'line', label: 'Line', format: 'integer' },
+    ],
+  },
+  {
+    name: 'UserTiming',
+    display: ['marker-chart', 'marker-table'],
+    data: [{ key: 'name', label: 'Name', format: 'string' }],
+  },
+  {
+    name: 'Text',
+    display: ['marker-chart', 'marker-table'],
+    data: [{ key: 'name', label: 'Name', format: 'string' }],
+  },
+  {
+    name: 'Log',
+    display: ['marker-table'],
+    data: [
+      { key: 'module', label: 'Module', format: 'string' },
+      { key: 'name', label: 'Name', format: 'string' },
+    ],
+  },
+  {
+    name: 'Bailout',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      { key: 'bailoutType', label: 'Type', format: 'string' },
+      { key: 'where', label: 'Where', format: 'string' },
+      { key: 'script', label: 'Script', format: 'string' },
+      { key: 'functionLine', label: 'Function Line', format: 'string' },
+      { key: 'bailoutLine', label: 'Bailout Line', format: 'string' },
+    ],
+  },
+  {
+    name: 'tracing',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      // This is really the "type" of the marker.
+      { key: 'category', label: 'Category', format: 'string' },
+    ],
+  },
+  {
+    name: 'IPC',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      { key: 'messageType', label: 'Type', format: 'string' },
+      { key: 'sync', label: 'Sync', format: 'string' },
+    ],
+  },
+];
+
+export function formatFromMarkerSchema(
```

Comment:
Nit: would be good to add a small comment on top of this function.

Reply:
Done!